src/Oscoin/Crypto/PubKey/RealWorld.hs

```diff
@@ -26,34 +25,39 @@ import           Oscoin.Crypto.PubKey.Internal (PK(..), SK(..))
 
 instance HasDigitalSignature Crypto where
 
+    -- N.b. The uses of 'data' instead of a 'newtype' are essential to avoid
```

Comment:
I think this warrants some further investigation - or an "authoritative" explanation of why this works, and that it is indeed safe (as opposed to deferring the problem to later).

Reply:
Totally -- let me open an issue for that.

Reply:
https://github.com/oscoin/oscoin/issues/525